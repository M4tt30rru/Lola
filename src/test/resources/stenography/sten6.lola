##Import "std.lola"
##Find(DefaultableArgumentDeclaration)
 ##Either ##ArgumentDeclaration(_dec) ##or ##ArgumentDeclaration(_dec) = ##Expression(_value)
##Find ##Identifier(_name) (##NoneOrMore ##DefaultableArgumentDeclaration(_dec) ##separator ,) = default;
 ##replace ##ForEach([i for i, d in enumerate(_decs) if hasattr(d, '_value')] + [len(_decs)]) ##(_name) (##(','.join(d._dec.name for d in _decs[:_]))) {##(''.join('this.' + d._dec._name.name + '=' + d._dec._name.name + ';' for d in _decs[:_])) ##(''.join('this.' + d._dec._name.name + '=' + d._value.name + ';' for d in _decs[_:]))}

A(int a, int b, int c=1, int d=a+b) = default;
