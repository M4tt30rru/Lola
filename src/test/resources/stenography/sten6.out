A(int a, int b) {
	this.a = a;
	this.b = b;
	this.c = 1;
	this.d = a + b;
}

A(int a, int b, int c) {
	this.a = a;
	this.b = b;
	this.c = c;
	this.d = a + b;
}

A(int a, int b, int c, int d) {
	this.a = a;
	this.b = b;
	this.c = c;
	this.d = d;
}

